### DISCUSSION TITLE
Detect unsed translations or missing ones globally

### DISCUSSION THREAD

**Discussion by existe-deja** [nuxt-modules/i18n#1187] [Ideas]
Hi,

How to detect unsed translations or missing ones globally?
For now, warnings only appear page by page. It's hard to be sure nothing is missing. Also, after a lot of refactoring, you can miss to clean the translation file.

I was thinking of global console warning or, maybe a cli tool. It can improve maintainability of large project a lot.

Does somebody have a workaround?

**Comment by rchl** [comment]
My answer to that is here (the first paragraph): https://github.com/nuxt-community/i18n-module/issues/1107#issuecomment-798475941

**Reply by existe-deja** [reply]
Thank you @rchl 

I think it's acceptable to force the way of writing translation keys if you want to use such a feature. Tailwind for exemple use [this approach](https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html) by basically saying "if you don't write something purgeable, it won't work."
It won't work for all cases but if you write down the translation key in plain text, it's possible to process it. Typically, if the keys come from a database it will not be covered.

Something I do is to nest my translations reproducing my file system. Ie: 
```
/components
   - AComp.vue
/pages
  - Index.vue
  /users
    - Index.vue
 ```
 
en.json
```
commons: {
  .. some reusable translation here
},
components: {
  "a-comp": {
      "foo": "bar"
   }
},
pages: {
  "index": {
      "title": "",
      "subtitle": ""
   },
  "users": {
      "index": {
         "title": "",
         "subtitle": ""
      }
   }
}
```

**Reply by existe-deja** [reply]
I just discovered https://github.com/pixari/vue-i18n-extract and it works pretty well as long as I don't use dynamic keys.

**Comment by rchl** [comment]
It might be best to leave that job to external tools like that. It's an additional tooling that is not strictly related to the core functionality. I think I wouldn't have time to maintain something like that anyway since it's all about heuristics and dealing with all kinds of input.
